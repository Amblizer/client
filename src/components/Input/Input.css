/* MESSANGER */

.input-messenger {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 48px;
  box-sizing: border-box;
  border: 0;
  border-right: 1px solid #cccccc;
  border-top: 1px solid #dddddd;
  background-color: #ffffff;
  padding: 8px;
  color: #222222;
  font-size: 11px;
  resize: none;
  outline: 0;
}

.input-messenger:disabled {
  background-color: #efefef;
}

.closed .input-messenger {
  display: none;
}

/* FLOAT */

.input-float {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 48px;
  box-sizing: border-box;
  box-shadow: #dddddd 1px 1px 8px 0;
  border: 0;
  background-color: #ffffff;
  border-radius: 10px;
  padding: 16px;
  color: #222222;
  font-size: 13px;
  resize: none;
  outline: 0;
}

.input-float:disabled {
  background-color: #e1e1e1;
}

.closed .input-float {
  display: none;
}

/* SIDEPANEL */

.input-sidepanel {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 48px;
  box-sizing: border-box;
  border: 0;
  border-top: 1px solid #dddddd;
  /*border-right: 1px solid #cccccc;*/
  background-color: #ffffff;
  padding: 8px;
  color: #222222;
  font-size: 11px;
  resize: none;
  outline: 0;
}

.input-sidepanel:disabled {
  background-color: #e1e1e1;
}

.closed .input-sidepanel {
  display: none;
}