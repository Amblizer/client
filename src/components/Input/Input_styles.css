.Input-input {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 48px;
  box-sizing: border-box;
  border: 0;
  border-right: 1px solid #ccc;
  border-top: 1px solid #ddd;
  background-color: #fff;
  padding: 8px;
  color: #222;
  font-size: 11px;
  resize: none;
  outline: 0;
}