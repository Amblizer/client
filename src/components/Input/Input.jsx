import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';

import './Input.css';

import { updateMessageList } from '../../containers/Chat/actions';


// TODO: move to a constants file?
const KEY_ENTER = 13;

export class InputComponent extends Component {
  static propTypes = {
    chatStyle: PropTypes.string,
    dispatch: PropTypes.func,
    socket: (props, propName) => {
      if (!(propName in props)) {
        throw new Error('socket must be set.');
      }
    },
  }


  constructor (props) {
    super(props);
    this.socket = props.socket;

    this.state = { messageBox: '' };
  }


  onKeyPress = (event) => {
    const { key, keyCode, shiftKey, ctrlKey, altKey } = event;
    console.log(`INPUT KEYPRESS ${key} (${keyCode}), SHIFT ${shiftKey}, CTRL ${ctrlKey}, ALT ${altKey}`);

    if (keyCode === KEY_ENTER) {
      if (!shiftKey) {
        console.log('SENDING MESSAGE ...');
        // Send data
        this.socket.emit('client:message', event.target.value);
        this.props.dispatch(updateMessageList(event.target.value));
        this.setState({ messageBox: '' });

        event.preventDefault();
      } else {
        // TODO: implement soft enter.
        // this.props.dispatch({ type: UI_SOFT_ENTER });
      }
    }

    this.socket.emit('client:typing');
  }

  handleChange = (e) => { this.setState({ messageBox: e.target.value }); }

  render() {
    const chatStyle = this.props.chatStyle;

    return (
      <textarea
        type="text"
        className={`Input_${chatStyle}`}
        placeholder="Type a message&hellip;"
        onKeyDown={this.onKeyPress}
        onChange={this.handleChange}
        value={this.state.messageBox}
      />
    );
  }
}

const mapStateToProps = state => ({ chatStyle: state.ui.chatStyle });
const mapDispatchToProps = dispatch => ({
  dispatch,
});

const Input = connect(
  mapStateToProps,
  mapDispatchToProps,
)(InputComponent);

export default Input;
