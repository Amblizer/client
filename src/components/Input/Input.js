import React, {Component} from 'react';
import {connect} from 'react-redux';
import './Input_styles.css'
import {updateMessageList} from '../../containers/Chat/actions'




// TODO: move to a constants file? 
const KEY_ENTER = 13;

class Input extends Component {
  static propTypes = {
    socket: (props, propName) => {
      if (!(propName in props)) {
        throw new Error('socket must be set.');
      }
    },
  }

  constructor (props) {
    super(props);
    this.socket = props.socket;
  }

  onKeyPress = (event) => {
    const {key, keyCode, shiftKey, ctrlKey, altKey, input} = event;
    console.log(`INPUT KEYPRESS ${key} (${keyCode}), SHIFT ${shiftKey}, CTRL ${ctrlKey}, ALT ${altKey}`);

    if (keyCode === KEY_ENTER) {
      if (!shiftKey) {
        console.log('SENDING MESSAGE ...');
        // Send data
        this.socket.emit('client:message', event.target.value);

        // Update message list
        this.props.dispatch(updateMessageList(event.target.value));
        // this.props.dispatch({ type: UI_HARD_ENTER, message: event.target.value });
        // this.props.dispatch({ type: CHAT_MESSAGE_CLIENT, message: event.target.value });

        event.preventDefault();
        // TODO: figure out how to deal with this?
        // redux form type stuff? Controlled input in component state?
        // input.value = '';
      } else {
        // Update input height
        this.props.dispatch({ type: UI_SOFT_ENTER });
      }
    }

    this.socket.emit('client:typing');
  }


  render() {
    return (
      <textarea
        type="text"
        className="Input-input"
        placeholder="Type a message&hellip;"
        onKeyDown={this.onKeyPress}
      />
    );
  }
}

const mapStateToProps = state => ({})
const mapDispatchToProps = dispatch => ({
  dispatch
})

export default connect(mapStateToProps, mapDispatchToProps)(Input);

// import React, { Component, PropTypes } from 'React';


// // @injectSheet(style('Input'))
// class Input extends Component {
//   static propTypes = {
//     socket: (props, propName) => {
//       if (!(propName in props)) {
//         throw new Error('socket must be set.');
//       }
//     },
//     // sheet: (props, propName) => {
//     //   if (!(propName in props)) {
//     //     throw new Error('sheet must be set.');
//     //   }
//     // },
//   }
//   constructor (props) {
//     super(props);

//     this.socket = props.socket;
//   }


//   // Event Handler

//   onKeyPress (event) {
//     const key = event.key;
//     const keyCode = event.keyCode;
//     const shiftKey = event.shiftKey;
//     const ctrlKey = event.ctrlKey;
//     const altKey = event.altKey;
//     const input = event.target;

//     console.log(`INPUT KEYPRESS ${key} (${keyCode}), SHIFT ${shiftKey}, CTRL ${ctrlKey}, ALT ${altKey}`);

//     if (keyCode === KEY_ENTER) {
//       if (!shiftKey) {
//         console.log('SENDING MESSAGE ...');
//         // Send data
//         this.socket.emit('client:message', event.target.value);

//         // Update message list
//         store.dispatch({ type: UI_HARD_ENTER, message: event.target.value });
//         store.dispatch({ type: CHAT_MESSAGE_CLIENT, message: event.target.value });

//         event.preventDefault();
//         input.value = '';
//       } else {
//         // Update input height
//         store.dispatch({ type: UI_SOFT_ENTER });
//       }
//     }

//     this.socket.emit('client:typing');
//   }


//   render () {
//     // const { sheet: { classes } } = this.props;

//     return (
//       <textarea
//         type="text"
//         className="Input-input"
//         placeholder="Type a message&hellip;"
//         onKeyDown={this.onKeyPress.bind(this)}
//       />
//     );
//   }
// }
  
// export default Input;