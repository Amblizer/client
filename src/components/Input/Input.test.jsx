import React from 'react';

import { shallow, mount } from 'enzyme';

import Input from './Input.jsx';

const socket = {
  // on: jest.fn(),
  emit: jest.fn(),
};

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({
    ui: { chatStyle: 'TEST' },
  })),
};

describe('Input', () => {
  afterEach(() => {
    socket.emit.mockReset();
  });

  it('matches snapshot', () => {
    const component = shallow(<Input store={store} socket={socket} />);
    expect(component).toMatchSnapshot();
  });

  it('emits a client:typing event on keyDown', () => {
    const component = mount(<Input store={store} socket={socket} />);
    component.find('textarea').simulate('keyDown', {
      keyCode: 84,
      key: 't',
    });

    expect(socket.emit).toHaveBeenCalled();
    expect(socket.emit).toHaveBeenCalledWith('client:typing');
  });

  it('emits a client:message event on keyDown', () => {
    const component = mount(<Input store={store} socket={socket} />);

    component.find('textarea').simulate('keyDown', {
      keyCode: 13,
      shiftKey: false,
      key: 'Enter',
    });

    expect(socket.emit).toHaveBeenCalledTimes(2);
    expect(socket.emit.mock.calls[0][0]).toBe('client:message');
  });
});
