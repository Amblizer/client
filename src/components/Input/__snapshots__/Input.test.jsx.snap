exports[`Input matches snapshot 1`] = `
<InputComponent
  chatStyle="TEST"
  dispatch={[Function]}
  socket={
    Object {
      "emit": [Function],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  } />
`;
